### PR TITLE
feat: Add step to install graphviz by default

### PR DESCRIPTION
**Pull request by m1ntySkye**
graphviz is very nice with doxygen. try it out with these doxy config options:

```
HAVE_DOT               = YES
CALL_GRAPH             = YES
CALLER_GRAPH           = YES
DOT_GRAPH_MAX_NODES    = 1000 # set this to be some high value that doesn't take too long
UML_LOOK               = YES
```

### CODE REVIEW

**Comment by DenverCoder1** [comment]
I haven't used graphviz with Doxygen, but do you think it's necessary that it be installed by default?

Another option could be to just add a step that installs it before running the Doxygen action (see #14).

**Comment by DenverCoder1** [comment]
If it's common enough to use it and it doesn't slow down the action noticably, I suppose it would be fine. Let me know what you think about this vs the other proposed solution.

**Comment by m1ntySkye** [comment]
I've found graphviz very useful, especially for someone who wants to just experiment with doxygen & wants to just see what it can do for a project with minimal setup (ie hasn't fully converted their comments to work with doxygen)

I had a look just now at two actions runs I had on my personal repo, one before and after making the action install graphviz by default. The timestamps seem to show apt-get taking 24s without graphviz & 16s with it. This makes me think that the added delay is minimal enough that it's not going to be differentiable from random processing delays in github action containers (unless I'm reading it wrong somehow)

an alternative but more complicated option would be to optionally install it based on if the user-provided doxyfile contains HAVE_DOT=YES 
ie `grep -i "HAVE_DOT\s*=\s*YES" ${{ inputs.config_file }}` (new to github actions, don't know the best way to do the rest of it)

**Comment by m1ntySkye** [comment]
just realised that grep wouldn't even work for my own repo, were I'm setting HAVE_DOT in an `@include`ed doxygen config file.

tldr; penalty of always installing seems pretty tiny

**Comment by DenverCoder1** [comment]
Other options we have could be to either:

* Have a boolean input option as to whether or not to install it (eg. `graphviz: true`)
* Have an input option with a list of additional dependencies to install (eg: `additional_deps: "graphviz"`)
* Document how to install `graphviz` using a `run` step _before_ the Doxygen GitHub action (eg. https://github.com/DenverCoder1/doxygen-github-pages-action/issues/14#issuecomment-1193372391)
* Search Doxyfile and included files for `HAVE_DOT`? (probably not so simple and may be slow)
* Always install `graphviz` which seems to not slow it down significantly (current option)

Of these options which do you think is best?

I think the current option is fine, so if you think it's the best option, let me know, and I'll merge it as it is.

**Review by DenverCoder1 — APPROVED** [review]

**Comment by DenverCoder1** [comment]
Thanks for the contribution! :tada:

**Comment by m1ntySkye** [comment]
Thanks for the approval! (sorry for not replying to previous message, got kinda busy)
